src/transformations/add-storage-gaps.ts

```diff
@@ -7,16 +7,44 @@ import { StorageLayout } from '../solc/input-output';
 import { Transformation } from './type';
 import { TransformerTools } from '../transform';
 
-// 100 slots of 32 contractSize each
-const TARGET_SIZE = 32 * 50;
+// By default, make the contract a total of 50 slots (storage + gap)
+const DEFAULT_SLOT_COUNT = 50;
+
+function* execall(re: RegExp, text: string) {
+  re = new RegExp(re, re.flags + (re.sticky ? '' : 'y'));
+  while (true) {
+    const match = re.exec(text);
+    if (match && match[0] !== '') {
+      yield match;
+    } else {
+      break;
+    }
+  }
+}
 
 export function* addStorageGaps(
   sourceUnit: SourceUnit,
   { getLayout }: TransformerTools,
 ): Generator<Transformation> {
   for (const contract of findAll('ContractDefinition', sourceUnit)) {
     if (contract.contractKind === 'contract') {
-      const gapSize = getGapSize(contract, getLayout(contract));
+      const doc =
+        typeof contract.documentation === 'string'
+          ? contract.documentation
+          : contract.documentation?.text ?? '';
+
+      let targetSlots = DEFAULT_SLOT_COUNT;
+      for (const { groups } of execall(
+        /^\s*(?:@(?<title>\w+)(?::(?<tag>[a-z][a-z-]*))? )?(?<args>(?:(?!^\s@\w+)[^])*)/m,
+        doc,
+      )) {
+        if (groups && groups.title === 'custom' && groups.tag === 'contract-size') {
+          targetSlots = parseInt(groups.args);
+        }
+      }
```

Comment:
Do you want to define a helper to extract all (custom?) natspec tags into an object, and then we can use that both here and when we check for overrides?